python/tests/test_user_session.py

```diff
@@ -0,0 +1,97 @@
+import base64
+import json
+from unittest import mock
+
+import pytest
+
+from numerous._client._graphql_client import GraphQLClient
+from numerous.generated.graphql.client import Client as GQLClient
+from numerous.user_session import Session
+
+
+class MockUser:
+    def __init__(self, user_info: dict[str, str], client: GraphQLClient) -> None:
+        self.id = user_info["user_id"]
+        self.name = user_info["name"]
+        self.client = client
+
+    @staticmethod
+    def from_user_info(user_info: dict[str, str], client: GraphQLClient) -> "MockUser":
+        """Create a MockUser instance from a user info dictionary."""
+        return MockUser(user_info, client)
+
+
+class MockCookieGetter:
```

Comment:
Another nit-pick. Generally in our codebases we would call this type of object a "Stub" rather than a mock. Mocks are sort more about testing the behaviour (e.g. did we call this function in the expected way), while a stub is purely about making the code work with pre-set test values :)

Reply:
Changed to stub

Reply:
I think you changed the user to be a stub, but not the cookie getter.